### ISSUE TITLE
Many display problems with Select2 4.0.1

### ISSUE THREAD

**Issue by michaelperrin** [select2/select2#3970]
Hello,

Since version 4.0.1 (no problem with version 4.0.1-rc1), even the simplest examples are not displayed correctly on all browsers (latest Chrome / Firefox / Safari at least): the dropdown is not displayed under the selector.

Here is an example: https://jsfiddle.net/michaelperrin/zxu8pxn3/1/ (clicking on the selector shows the dropdown, but it is displayed too much on the left).

**Comment by kevin-brown** [comment]
This appears to be a bug when there is a margin set on the body and the `position` is `static`.

So this is definitely related to https://github.com/select2/select2/commit/c9216b4b966653dd63a67e815b47899ef5325298, because it only affects statically positioned elements.

**Comment by jonahgreenthal** [comment]
I had a possibly similar issue in which the dropdown portion appeared slightly lower and to the right of where it should have been. Do you think this is related? If so, what additional information, if any, would be helpful?

For the time being I just papered it over by tweaking the positioning in my custom CSS overrides, but obviously I'd prefer a better solution.

**Comment by kevin-brown** [comment]
For anyone running into this issue, it would be useful to have the output for the following two statements (executed in your browser console, F12 usually):

```js
document.body.style.position
```
and 
```js
$(document.body).offset()
```

On the documentation website, I get the following output

![image](https://cloud.githubusercontent.com/assets/1991850/11461982/3b49dd3a-96db-11e5-95c9-235af08936a6.png)

Which explains why the dropdown is lining up fine there.

**Comment by jonahgreenthal** [comment]
I misremembered -- it's slightly above and left of where it should be. Anyway, Here's the console output from those commands:

```
> document.body.style.position
< ""
> $(document.body).offset()
< Object { top=8,  left=8}

**Comment by svivian** [comment]
I'm getting the same issue. Here's the output:

    > document.body.style.position
      ""
    > $(document.body).offset()
      Object {top: 20, left: 20}

**Comment by BlaM** [comment]
Same here...

    > document.body.style.position
    < ""
    > $(document.body).offset()
    < Object {top: 65, left: 0}

And yes, it's positioned 65px too high, "left" position is correct.

**Comment by eric-poitras** [comment]
Same problem: 
```
$(document.body).offset()
Object {top: 54, left: 0}
```

**Comment by vincentwoo** [comment]
For me, 4.0.1 opens the dropdown and immediately closes it whenever I click on the select. Reverting to 4.0.0 fixes the problem.

**Comment by kevin-brown** [comment]
> For me, 4.0.1 opens the dropdown and immediately closes it whenever I click on the select. Reverting to 4.0.0 fixes the problem.

This is probably a separate, unrelated issue. If you can create a jsbin/jsfiddle that reproduces the issue and a new ticket describing the issue, we can handle the bug.

**Comment by vincentwoo** [comment]
It seems like the same issue. Also seems to be caused by CSS problems. Here's a fiddle: https://jsfiddle.net/hacf7fdy/

**Comment by BlaM** [comment]
It's the same issue. Had the same thing here and it's caused by having one of the list items at the location of the mouse as soon as the drop down opens and when you "mouseup" select2 thinks the item has been selected and closes.

**Comment by kevin-brown** [comment]
> It seems like the same issue.

Ah, you're right. I forgot how we handle events in the dropdown.

The dropdown is getting placed beneath the mouse, and the `mouseup` event is being fired on it so it is closing.

**Comment by LordSputnik** [comment]
Same here - noticed this because of an item being selected when the dropdown opens, as other people seem to have found.

    > document.body.style.position
    < ""
    > $(document.body).offset()
    < Object {top: 64, left: 0}

**Comment by Salubritas** [comment]
Same here. Dropdown is appearing too high.

```
document.body.style.position
""
$(document.body).offset()
Object { top: 10, left: 0 }
```

**Comment by ramitayba** [comment]
same issue appear with fixed header template

**Comment by kevin-brown** [comment]
Note that this issue should be fixed with the latest master if you manually recompile the assets. It will be included in the upcoming 4.0.2-rc.1 release.
